core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -31,7 +38,9 @@ import scala.util.{Failure, Success, Try}
 /**
   * Created by chris on 12/26/15.
   */
-sealed abstract class ScriptPubKey extends Script
+sealed abstract class ScriptPubKey extends Script {
+  def pubKeysFor(inputInfo: InputInfo): Vector[ECPublicKey]
```

Comment:
I think it'd make more sense to just have a `ScriptPubKey.pubKeys` and then put all the invariants inside of the `InputInfo`s

Reply:
You might even be able to create a cleaner interface to use if you move it into companion objects: https://github.com/Christewart/bitcoin-s-core/tree/2020-05-spk-refactor-suggestion